src/components/App/DisplayPuzzleInstances/index.tsx

```diff
@@ -0,0 +1,137 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+import styles from './DisplayPuzzleInstances.module.scss';
+import { PuzzleCustom } from '../../../types/api/puzzles/puzzle';
+import { getAllPuzzleInstances } from '../../../services';
+import { PuzzleInstance } from '@prisma/client';
+import { Button } from '../../Global';
+import { QRGenerator } from '../../App';
+import toast from 'react-hot-toast';
+
+type DisplayPuzzleInstancesProps = {
+  puzzlesList: PuzzleCustom[];
+};
+
+const DisplayPuzzleInstances = ({
+  puzzlesList
+}: DisplayPuzzleInstancesProps) => {
+  const [puzzleId, setPuzzleId] = useState<string>('');
+  const [puzzleInstances, setPuzzleInstances] = useState<PuzzleInstance[]>([]);
+  const [displayPuzzleInstances, setDisplayPuzzleInstances] = useState<
+    PuzzleInstance[]
+  >([]);
+
+  const getPuzzleInstances = async () => {
+    const retrivedPuzzleInstances = await getAllPuzzleInstances();
+    setPuzzleInstances(retrivedPuzzleInstances);
+  };
+
+  useEffect(() => {
+    getPuzzleInstances();
```

Comment:
Is this effect hook necessary when we have the second one below?

Reply:
Nice catch

Reply:
👌